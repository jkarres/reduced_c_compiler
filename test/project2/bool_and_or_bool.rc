function : bool t() { cout << "in t()" << endl; return true; }
function : bool f() { cout << "in f()" << endl; return false; }

function : void main() {

  cout << 1 << " " << true && true << endl;
  cout << 2 << " " << true && false << endl;
  cout << 3 << " " << true && t() << endl;
  cout << 4 << " " << true && f() << endl;
  cout << 5 << " " << false && true << endl;
  cout << 6 << " " << false && false << endl;
  cout << 7 << " " << false && t() << endl;
  cout << 8 << " " << false && f() << endl;
  cout << 9 << " " << t() && true << endl;
  cout << 10 << " " << t() && false << endl;
  cout << 11 << " " << t() && t() << endl;
  cout << 12 << " " << t() && f() << endl;
  cout << 13 << " " << f() && true << endl;
  cout << 14 << " " << f() && false << endl;
  cout << 15 << " " << f() && t() << endl;
  cout << 16 << " " << f() && f() << endl;
  cout << 17 << " " << true || true << endl;
  cout << 18 << " " << true || false << endl;
  cout << 19 << " " << true || t() << endl;
  cout << 20 << " " << true || f() << endl;
  cout << 21 << " " << false || true << endl;
  cout << 22 << " " << false || false << endl;
  cout << 23 << " " << false || t() << endl;
  cout << 24 << " " << false || f() << endl;
  cout << 25 << " " << t() || true << endl;
  cout << 26 << " " << t() || false << endl;
  cout << 27 << " " << t() || t() << endl;
  cout << 28 << " " << t() || f() << endl;
  cout << 29 << " " << f() || true << endl;
  cout << 30 << " " << f() || false << endl;
  cout << 31 << " " << f() || t() << endl;
  cout << 32 << " " << f() || f() << endl;

}
