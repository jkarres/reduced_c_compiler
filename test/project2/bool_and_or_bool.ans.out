1 true
2 false
3 in t()
true
4 in f()
false
5 false
6 false
7 false
8 false
9 in t()
true
10 in t()
false
11 in t()
in t()
true
12 in t()
in f()
false
13 in f()
false
14 in f()
false
15 in f()
false
16 in f()
false
17 true
18 true
19 true
20 true
21 true
22 false
23 in t()
true
24 in f()
false
25 in t()
true
26 in t()
true
27 in t()
true
28 in t()
true
29 in f()
true
30 in f()
false
31 in f()
in t()
true
32 in f()
in f()
false
